docker/README.md

```diff
@@ -1,56 +1,60 @@
 
-Now you can start up the entire Nginx + PHP + MySQL stack using docker with:
 
-    $ docker-compose up -d
+## Environment Dependencies
 
-the `-d` flag will launch this in the background so you can continue to use the terminal window. On the first run you need to init the database (your container name may be different depending on the name of your root directory):
+Though Docker will take care of this for the most part, there are 2 things you'll need to get started.
 
-    $ docker exec -it -u www-data userfrosting_php_1 sh -c 'php bakery migrate'
+* [Docker Compose](https://docs.docker.com/compose/install/)
+* Bash plus the common Linux tools (for Windows users, Git Bash will cover this)
 
-You also need to setup the first admin user (again, your container name may be different depending on the name of your root directory):
+## First Run
 
-    $ docker exec -it -u www-data userfrosting_php_1 sh -c 'php bakery create-admin'
+>Certain `bakery` commands such as `bake` that use NodeJS are not currently supported but should work provided a suitable image is used.
 
-Now visit `http://localhost:8591/` to see your UserFrosting homepage!
+Run the following through `bash` from the project root directory;
 
-**Paste these into a bash file and execute it!**
+```bash
+# Create ".env" and "sprinkles.json" from examples
+cp app/sprinkles.example.json app/sprinkles.json
+cp app/.env.example app/.env # This will need to be updated to reflect docker-compose.yml
 
-```
+# Install composer dependencies
+docker run --rm -it -v `pwd -W || pwd`:/app composer update  --ignore-platform-reqs --no-scripts
+# Alternative (deprecated): docker-compose run composer install --ignore-platform-reqs --no-scripts
+
+# Install npm dependencies
+docker run --rm -it -v `pwd -W || pwd`:/app node:alpine sh -c "cd /app/build ; npm install ; npm run uf-assets-install"
+# Alternative (deprecated): docker-compose run node  sh -c "cd /app/build ; npm install ; npm run uf-assets-install"
+
+# Prepare folder permissions
 chmod 777 app/{logs,cache,sessions}
-docker-compose build --force-rm --no-cache
-docker-compose run composer install --ignore-platform-reqs --no-scripts
-docker-compose run node npm install
-docker-compose run composer update --ignore-platform-reqs --no-scripts
-docker-compose run node npm run uf-assets-install
+
+# Bring up the stack
 docker-compose up -d
-echo -n "Enter Docker Container Name --> "
-read docker_container
-docker exec -it -u www-data $docker_container sh -c 'php bakery migrate'
-docker exec -it -u www-data $docker_container sh -c 'php bakery create-admin'
-```
 
-**This is not (yet) meant for production!**
+# Capture container name
+docker_container=$(docker ps -q --filter="ancestor=userfrosting_php")
 
-You may be tempted to run with this in production but this setup has not been security-hardened. For example:
+# Perform database migration
+docker exec -it -u www-data $docker_container sh -c "php bakery migrate"
```

Comment:
The problem here (as I mentioned in rocketchat) is that the documentation says to use --skip-scripts, which is used to pull in a bunch of other things.  This is why you have to run composer twice.  There's no real way to fix this without building the container properly BEFORE running composer in it.

Reply:
`--skip-scripts` being removed was an inconsistent fix when I was testing this. I've updated the docs to reflect the general flakiness of it, so that it should an issue come up people have options.

Changes I have in mind for the Docker configuration should improve reliability later on.